src/Data/JoinSemilattice/Class/Ord.hs

```diff
@@ -12,43 +14,43 @@ License     : MIT
 module Data.JoinSemilattice.Class.Ord where
 
 import Control.Applicative (liftA2)
-import Data.Hashable (Hashable)
 import Data.JoinSemilattice.Defined (Defined (..))
-import Data.JoinSemilattice.Intersect (Intersect (..))
+import Data.JoinSemilattice.Intersect (Intersect (..), Intersectable)
 import qualified Data.JoinSemilattice.Intersect as Intersect
 import Data.JoinSemilattice.Class.Boolean (BooleanR (..))
-import Data.JoinSemilattice.Class.Eq (EqR)
-import Data.Kind (Type)
+import Data.Kind (Constraint, Type)
 
 -- | Comparison relationships between two values and their comparison result.
-class EqR x b => OrdR (x :: Type) (b :: Type) | x -> b where
+class OrdR (f :: Type -> Type) (c :: Type -> Constraint) | f -> c where
```

Comment:
I couldn't keep the `EqR` superclass on first try, due to the `Defined` instance: I have `EqR Defined Eq` but `OrdR Defined Definable`. This might be fixed by defining `EqR Defined Definable` instead, but the `Ord` seems undesirable in there. Or some kind of

```
class EqR f c => OrdR f c' | f -> c, f -> c' where
  ...
```

Or some way to map between `c` and `c'`?

Reply:
(Or just leave the superclass out as it is in this version.)

Reply:
Here's another approach that kind of sort of maybe helps here:

```
class EqR (f :: Type -> Type) where
  type CEq f :: Type -> Constraint
  eqR :: (CEq f) x => ( f x, f x, f Bool ) -> ( f x, f x, f Bool )

class EqR f => OrdR (f :: Type -> Type) where
  type COrd f :: Type -> Constraint
  lteR :: (COrd f) x => ( f x, f x, f Bool ) -> ( f x, f x, f Bool )

instance EqR Defined where
  type CEq Defined = Eq
  eqR (x, y, z) = ...

class (Eq x, Ord x) => Definable x
instance (Eq x, Ord x) => Definable x

instance OrdR Defined where
  type COrd Defined = Definable
  lteR (x, y, z) = ...
```

I feel like there should be some assertion in here that `COrd f x` implies `CEq f x`, but I can't quite figure out how to express that. `CEq f = X . COrd f` on a type level.

Reply:
Honestly? I'd just put an `Ord` constraint on the `EqR` instance of `Defined` and be done with it. It's not perfect, but this seems like a much easier compromise

Reply:
Sure! I've just pushed a commit to do that. Not sure there's any benefit to the change though tbh.